scripts/compose-update-allowlist-tx.py

```diff
@@ -23,43 +94,33 @@ def get_entity_ids(p):
         entity = open(fullpath)
         entity_json = json.load(entity)
         entity_id.append(entity_json['OasisEntityID'])
-    return ','.join(entity_id)
+    return entity_id
 
 def print_tx(args):
+    runtime_descriptor = json.loads(runtime_descriptor_json)
+    for entity in get_entity_ids(args.network):
+        runtime_descriptor['admission_policy']['entity_whitelist']['entities'][entity] = {}
+    with open(args.runtime_descriptor, 'w') as f:
+        f.write(json.dumps(runtime_descriptor, sort_keys=True, indent=2))
+
     output = f"""
-    export ENTITY_DIR={args.entity_dir}
-    export ENTITY_ID={get_entity_ids(args.network)}
-    export GENESIS_JSON={args.genesis_json}
-    export RUNTIME_ID={args.runtime_id}
-    export NONCE={args.nonce}
-    cd {args.data_dir}
-    ../oasis-node registry runtime gen_register -y \\
-      --transaction.fee.gas 10000 \\
-      --transaction.fee.amount 0 \\
-      --transaction.file register_runtime.tx \\
-      --transaction.nonce $NONCE \\
-      --genesis.file $GENESIS_JSON \\
-      --signer.backend file \\
-      --signer.dir $ENTITY_DIR \\
-      --runtime.id $RUNTIME_ID \\
-      --runtime.kind compute \\
-      --runtime.executor.group_size 3 \\
-      --runtime.executor.group_backup_size 5 \\
-      --runtime.executor.allowed_stragglers 1 \\
-      --runtime.storage.group_size 2 \\
-      --runtime.storage.min_write_replication 2 \\
-      --runtime.admission_policy entity-whitelist \\
-      --runtime.admission_policy_entity_whitelist $ENTITY_ID \\
-      --runtime.genesis.state ../etc/oasis_genesis_mainnet_testing_ff03.json \\
-      --runtime.txn_scheduler.flush_timeout 1s \\
-      --runtime.txn_scheduler.max_batch_size 10000 \\
-      --runtime.txn_scheduler.max_batch_size_bytes 16mb \\
-      --runtime.storage.checkpoint_chunk_size 16777216 \\
-      --runtime.storage.checkpoint_interval 1000 \\
-      --runtime.storage.checkpoint_num_kept 5
-    ../oasis-node consensus submit_tx --transaction.file register_runtime.tx
-    rm -f register_runtime.tx
-    """
+cd {args.data_dir}
+export GENESIS_JSON={args.genesis_json}
+export NONCE={args.nonce}
+export ENTITY_DIR={args.entity_dir}
+export RUNTIME_DESCRIPTOR={args.runtime_descriptor}
+../oasis-node-new registry runtime gen_register -y \\
```

Comment:
Because our validator does not upgrade to v21.1.2, I use `oasis-node-new` here for version v21.1.2.

Reply:
Please rename it to `oasis-node-v21.1.2` to make sure the version is correct. `new` is not enough to identify the version number.

Reply:
Updated.